examples/contrib/readhar.py

```diff
@@ -0,0 +1,134 @@
+
+        requestHeaders = self.fix_headers(request_json["request"].get("headers", []))
+
+        client_conn = connection.Client(
+            peername=("127.0.0.1", 51513),
+            sockname=("127.0.0.1", 8080),
+            timestamp_start=time.time(),
+        )
+        server_conn = connection.Server(address=None)
+
+        newflow = http.HTTPFlow(client_conn, server_conn)
+        try:
+            newflow.request = http.Request.make(
+                requestMethod, requestUrl, "", requestHeaders
+            )
+        except:
+            logging.error("Unable to create flow, please change to a valid HAR format")
+            return
+
+        responseCode = request_json["response"].get("status", "")
+
+        responseContent = request_json["response"]["content"].get("text", "")
+
+        responseHeaders = self.fix_headers(request_json["response"].get("headers", []))
+        try:
+            newflow.response = http.Response.make(
+                responseCode, responseContent, responseHeaders
+            )
+        except:
+            logging.error("Unable to create flow, please change to a valid HAR format")
+            return
+
+        return newflow
+
+    async def loadFlows(self) -> None:
+        """
+        Loads flows into mitmproxy
+
+        Raises:
+            exceptions.FlowReadException: If unable to load flow the FlowReadException is raised
+        """
+        cnt = 0
+        for flow in self.flows:
+            try:
+                await ctx.master.load_flow(flow)
+                cnt += 1
+            except (OSError, exceptions.FlowReadException) as e:
+                if cnt:
+                    logging.warning("Flow file corrupted - loaded %i flows." % cnt)
+                else:
+                    logging.error("Flow file corrupted.")
+                raise exceptions.FlowReadException(str(e)) from e
+
+    def running(self):
+        """
+        Runs the loadFlows function
+        """
+        self._read_har_task = asyncio.create_task(self.loadFlows())
+
+    @command.command("readhar")
+    def readhar(
+        self,
+        path: types.Path,
+    ) -> None:
+        """
+        Reads a HAR file into mitmproxy. Loads a flow for each entry in given HAR file.
+
+        Args:
+            path (types.Path): Path to HAR file
+        """
+        with open(path) as fp:
+            try:
+                self.harFile = json.load(fp)
+            except:
+                try:
+                    json.load(codecs.open(path, "r", "utf-8-sig"))
+                except:
+                    logging.error(
+                        "Unable to read HAR file. Please provide a valid HAR file"
+                    )
+                    return
+
+        for request_json in self.harFile["log"]["entries"]:
+            flow = self.request_to_flow(request_json)
+            if flow:
+                self.flows.append(flow)
+        self.running()
+        self._read_har_task
```

Comment:
This here can be greatly simplified I think. You can just do this here instead and avoid the functions above:
```suggestion
        async def load_flows() -> None:
            for flow in flows:
                await ctx.master.load_flow(flow)
        asyncio.create_task(load_flows())
```

The main underlying problem here is that commands cannot be async yet, otherwise we could just call `await ctx.master.load_flow(flow)` directly.